src/Item.java

```diff
@@ -0,0 +1,45 @@
+public interface Item {
```

Comment:
ta klasa wygląda bardzo dobrze, zazwyczaj interfejs zaczyna się na I, tutaj śmiesznie bo IItem by było xD
Dobre nazwy dla interfejsów to też coś co opisuje ich funkcję (dla mnie Item brzmi jak klasa bazowa)

Np
IHasPosition

Reply:
Chociaż w sumie teraz ogarnąłem że w Javie interfejsy mają logikę biznesową, nie wiem czy tak się teraz robi czy nie, bardzo możliwe że tak.

Kiedyś to była jedna z różnic pomiędzy interfejsem a klasą bazową

Reply:
Więc co do nazw nie słuchaj mnie tylko sprawdź jak jest w Javie